pkg/vsphere/tasks/waiter.go

```diff
@@ -56,20 +57,21 @@ func Wait(ctx context.Context, f func(context.Context) (Task, error)) error {
 //       return vm, vm.Reconfigure(ctx, config)
 //    })
 func WaitForResult(ctx context.Context, f func(context.Context) (Task, error)) (*types.TaskInfo, error) {
-	var err error
 	var info *types.TaskInfo
 	var backoffFactor int64 = 1
 
 	for {
-		var t Task
-		if t, err = f(ctx); err == nil {
+		t, err := f(ctx)
+		log.Debugf("New task: %s", taskMoid(t))
+
+		if err == nil {
 			info, err = t.WaitForResult(ctx, nil)
```

Comment:
This is very strange.  My original PR (which was squashed and merged) had a check in `taskMoid` for nil.  I sqaushed and merged *with this check!*

But, the revert doesn't include this code.  So 1-  PRs are not immuteable once they are commited since my revert of the revert now includes this check and 2- squash and merge is totally broken!